torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1157,8 +1161,8 @@ void ProcessGroupNCCL::shutdown(std::optional<std::string> reason) {
   // lauch abort asynchrounously and wait for it to complete or timeout
   LOG(INFO) << logPrefix()
             << "Launching ProcessGroupNCCL abort asynchrounously.";
-  std::future<bool> fut = std::async(
-      std::launch::async, [this, &reason]() { return this->abort(reason); });
+  std::future<bool> fut =
+      std::async(std::launch::async, [this]() { return this->abortComms(); });
```

Comment:
We should never need this async calls for abortComms in the future? maybe add a TODO comment to remove the async call